ui/core/components/suggest_reforges_action.tsx

```diff
@@ -525,24 +509,26 @@ export class ReforgeOptimizer {
 									values: [
 										{ name: 'Select preset', value: 0 },
 										...[...statPresets.keys()].map(key => {
-											const ratingValue = statPresets.get(key)!;
-											const percentOrPointsValue = unitStat.convertRatingToPercent(ratingValue)!;
-											const percentValue = unitStat.equalsStat(Stat.StatMasteryRating) ? (percentOrPointsValue * this.player.getMasteryPerPointModifier()) : percentOrPointsValue;
+											const percentOrPointsValue = statPresets.get(key)!;
+											const percentValue = unitStat.equalsStat(Stat.StatMasteryRating)
+												? percentOrPointsValue * this.player.getMasteryPerPointModifier()
+												: percentOrPointsValue;
+
 											return {
 												name: `${key} - ${percentValue.toFixed(2)}%`,
-												value: ratingValue,
+												value: percentOrPointsValue,
 											};
 										}),
 									].sort((a, b) => a.value - b.value),
 									enableWhen: () => this.isAllowedToOverrideStatCaps || !this.softCapsConfig.some(config => config.unitStat.equals(unitStat)),
 									getValue: () => {
-										let ratingValue = unitStat.convertDefaultUnitsToRating(this.statCaps.getUnitStat(unitStat))!;
+										let ratingValue = this.statCaps.getUnitStat(unitStat);
 										if (unitStat.equalsStat(Stat.StatMasteryRating)) ratingValue = this.toVisualBaseMasteryRating(ratingValue);
 
 										return ratingValue;
 									},
 									setValue: (_eventID, _player, newValue) => {
-										const statValue = unitStat.equalsStat(Stat.StatMasteryRating) ? (newValue + this.baseMastery) : unitStat.convertRatingToDefaultUnits(newValue)!;
+										const statValue = unitStat.equalsStat(Stat.StatMasteryRating) ? newValue + this.baseMastery : newValue;
```

Comment:
Same thing here, you need to copy the logic from the `percentagePicker` so it will work for both Stats and PseudoStats and automatically convert units.

Reply:
In fact, if possible, try to re-use all the same code in `percentagePicker` for the presets picker rather than copy-pasting everything.